Framework/PythonInterface/mantid/api/src/Exports/IFunction.cpp

```diff
@@ -35,6 +36,32 @@ PyObject *getCategories(IFunction &self) {
   return registered;
 }
 
+/// An object for constructing a shared_ptr that won't ever delete its pointee
+class NoDeleting {
+public:
+  /// Does nothing
+  void operator()(void *) {}
+  /// Does nothing
+  void operator()(const void *) {}
+};
+
+//------------------------------------------------------------------------------------------------------
+/**
+* Something that makes IFunction return to python a composite function
+* for Product function, Convolution or
+* any similar superclass of composite function.
+* @param self :: Enables it to be called as a member function on the
+* IFunction class
+*/
+Mantid::API::CompositeFunction_sptr castToCompositeFunction(IFunction &self) {
```

Comment:
@mantid-roman I thought about what ``__len__()`` should return for a plain FunctionWrapper and found that the following script works:
```
g = Gaussian(Sigma=.5,Height=3)
print g
print g[0], g[1], g[2]
g[1] = 7
print g
```
and therefore ``__len__()`` should return the number of parameters and this is very easy to implement.  I'm still to think of other 'composite' methods.